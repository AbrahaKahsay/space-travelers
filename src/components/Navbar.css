.navbar {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-items: flex-start;
  gap: 10%;
}

.logo {
  height: 60px;
  width: 60px;
}

.logo-img {
  width: 100%;
  height: 100%;
}

.links ul {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 1rem;
}

.links ul li {
  text-decoration: none;
  list-style: none;
  color: orange !important;
}
