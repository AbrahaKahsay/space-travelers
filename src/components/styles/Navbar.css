@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@300&display=swap');

* {
  box-sizing: border-box;
  margin: 0;
}

.profile::before {
  content: '|';
  margin-right: 1.2rem;
  font-weight: bold;
  color: #333;
}

.underline {
  text-decoration: underline;
  color: #4169e1;
}

.nav {
  display: flex;
  padding: 1rem;
  padding-left: 8rem;
  border-bottom: 1px solid gray;
}

.nav-items > * {
  display: inline-block;
  padding: 0.5rem;
  font-size: 1.2rem;
}

ul.nav-container-items {
  color: #4169e1;
}

.logo {
  margin-right: auto;
}

.logo p {
  font-family: 'Roboto', sans-serif;
  font-weight: 700;
  font-size: 2rem;
  width: 100%;
  padding-left: 5rem;
}

.logo-container {
  display: flex;
  align-items: center;
}

.logo-container img {
  height: 60px;
  width: 4rem;
}

.nav-container {
  align-items: center;
  padding: 1rem 10%;
}

.nav-container ul {
  font-family: 'Roboto', sans-serif;
  display: flex;
  gap: 2rem;
  list-style: none;
  align-items: center;
  text-decoration: none;
}
