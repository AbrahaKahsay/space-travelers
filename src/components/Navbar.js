import './Navbar.css';
import { NavLink } from 'react-router-dom';
import Logo from '../components/planet.png';

const Navbar = () => (
    <nav className="navbar">
        <div className='logo-container'>
            <div><img className='logo' src={Logo} alt="Logo" /></div>
            <h1>{ "Space Travelers' Hub"} </h1>
        </div>
        <div className="links">
           <ul>
            <li><NavLink className="link-list" to="/">Rockets</NavLink></li>
            <li><NavLink className="link-list" to="/mission">Mission</NavLink></li>
            <li><NavLink className="link-list" to="/profile">My Profile</NavLink></li>
           </ul> 
        </div>
    </nav>
);

export default Navbar;