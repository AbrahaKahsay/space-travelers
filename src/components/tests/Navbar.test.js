import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Navbar from '../../components/Navbar';

describe('Jest Snapshot testing suite', () => {
  it('Matches DOM Snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});