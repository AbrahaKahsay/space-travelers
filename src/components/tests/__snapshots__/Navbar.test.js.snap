// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<nav
  class="navbar"
>
  <div
    class="logo-container"
  >
    <div>
      <img
        alt="Logo"
        class="logo"
        src="planet.png"
      />
    </div>
    <h1>
      Space Travelers' Hub
       
    </h1>
  </div>
  <div
    class="links"
  >
    <ul>
      <li>
        <a
          aria-current="page"
          class="link-list active"
          href="/"
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          class="link-list"
          href="/mission"
        >
          Missions
        </a>
      </li>
      <li>
        <a
          class="link-list"
          href="/profile"
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
